docs/Practices/hub/hub.md

```diff
@@ -0,0 +1,118 @@
+
+- A potential iteration to reduce coupling between slices in routing is to have each slice directory be able to define it's own unique route while having a shared layer at the bounded context root then reference those definitions. There are many implementations that could achieve this. One could be to use nested routing where each slice/feature set (we don't utilize feature sets yet) would define it's own router and wire handlers and then the bounded context root would aggregate those routers underneath it's own router. Another option is to have each slice define its own unique path constant and have bounded context root simply reference those constants while wiring. 
+
+## Recommendations
+These recommendations, while value adding, do not necessitate being a guideline and being subject to the remediation requirements of a guideline. They are worth understanding and applying as fit.
+
+- Validate at all IO boundaries (guideline above indicates just handler request data). *Typescript can be a powerful tool, but it is only useful at compile time (as you write and build code). Validating that data being ingested by a process written using Typescript can go a long way in ensuring that the type safety provided by Typescript is reliable. Omitting validation at a boundary leaves room for undesirable behavior and sneaky errors (imagine getting "cannot read property X of undefined" when the type system supposedly doesn't allow it)*
+
+- Endeavour to make invalid states impossible. [Examples](./examples.md#endeavour-to-make-invalid-states-impossible)
+
+## Wiggle Room
+This document doesn't aim to define code styling (commands, linting rules, args as object vs not) nor does it aim to define the implementation of given abstractions as long as they don't deviate from the guidelines. *For example you could have one Vertical Slice utilizing a driver to execute queries directly against a database through a connection while another Vertical Slice utilizes an ORM. As long as both slices don't expose database concerns to the consumer of a given slice, then both strategies are acceptable. Another example would be tests. Maybe one slice tests itself by unit tests via isolated dependency and mocks + integration tests for side effects, while another test just uses api level integration (acceptance) testing. Both are acceptable as long as there is coverage.*
+
+## Contributing
+TODO
+
+### Process to change
+TODO
+
+### Tips for changes
+Guidelines:
+- Make sure a guideline implements the intent of a UA - whether it's a new or existing one.
```

Comment:
UA?

Reply:
Maybe link to the Useful Abstraction definition.

Reply:
linked to def